optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/drools/DroolsScoreDirector.java

```diff
@@ -82,6 +82,9 @@ private void resetKieSession() {
         }
     }
 
+    // The scoreHolder method is not actually deprecated for removal.
+    // It will only be moved to a different type at a time when we can make that change in public API.
+    @SuppressWarnings("deprecation")
```

Comment:
Would it be possible to prefix such comments with `// TODO remove in 8.0`?
The reason I like the "8.0" there is because we'll do a find in path for "8.0" after branching